extensions/omniv21/fileformat/edi/reader.go

```diff
@@ -0,0 +1,104 @@
+package edi
+
+import (
+	"bufio"
+	"fmt"
+
+	"github.com/jf-tech/omniparser/idr"
+)
+
+type rawSegElem struct {
+	elemIndex int // for this piece of data, the index of which element it belongs to. 1-based.
+	compIndex int // for this piece of data, the index of which component it belongs to. 1-based.
+	data      []byte
+}
+
+type rawSeg struct {
+	name  string       // name of the segment, e.g. 'ISA', 'GS', etc.
+	raw   []byte       // the raw data of the entire segment, including segment delimiter.
+	elems []rawSegElem // all the broken down pieces of elements of the segment.
+}
+
+const (
+	defaultElemsPerSeg = 20
```

Comment:
why we need this default value?

Reply:
In our experience, we've never seen more than 20 elements in a single segment, thus we can pre-allocate (see `newRawSeg()` down below)  the slice with `defaultElemsPerSeg` so that for the entire reading EDI process, we never need make another allocation.

Same reason for `defaultStackDepth`. Each stack level corresponds to one level in nested segments. Such `ISA->GS->LX->LS->MAN` that's *deep* nested seg stack is only 5 level deep. In fact now I'm thinking about we can reduce the default stack size to 10 is probably way more than enough.

In both cases, if some rare situation the size exceeds the default value, the slices would simply grow, not a big deal.